controlplane/kubeadm/internal/control_plane.go

```diff
@@ -222,3 +226,24 @@ func (c *ControlPlane) NeedsReplacementNode() bool {
 func (c *ControlPlane) HasDeletingMachine() bool {
 	return len(c.Machines.Filter(machinefilters.HasDeletionTimestamp)) > 0
 }
+
+// ProvisioningMachines returns machines that are still booting.  In the case
+// of 3 node or larger clusters, it excludes unhealthy machines.
+func (c *ControlPlane) ProvisioningMachines() FilterableMachineCollection {
```

Comment:
What about adding also `HasProvisioningMachines` and `HasUnhealthyMachines` so we are consistent with `HasDeletingMachine` and we can clean up some test above from `len(...)>0`

Reply:
I got push back for refactoring before so I didn't want to introduce more methods than necessary